spec.html

```diff
@@ -18,6 +18,110 @@ <h1>Array Objects</h1>
+                  1. <ins>Set _j_ to _j_ + 1.</ins>
+                1. <ins>Repeat, while _j_ &lt; _len_,</ins>
+                  1. <ins>Perform ? DeletePropertyOrThrow(_obj_, ! ToString(𝔽(_j_))).</ins>
+                  1. <ins>Set _j_ to _j_ + 1.</ins>
+                1. <ins>Return _obj_.</ins>
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-comparearrayelements" type="abstract operation">
+                <h1>
+                  CompareArrayElements(
+                    _x_: unknown,
+                    _y_: unknown,
+                    _comparefn_: a function object or *undefined*
+                  ): either a normal completion containing a Number or an abrupt completion
+                </h1>
+                <dl class="header">
+                </dl>
+                <emu-alg>
+                    1. If _x_ and _y_ are both *undefined*, return *+0*<sub>𝔽</sub>.
+                    1. If _x_ is *undefined*, return *1*<sub>𝔽</sub>.
+                    1. If _y_ is *undefined*, return *-1*<sub>𝔽</sub>.
+                    1. If _comparefn_ is not *undefined*, then
+                        1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
+                        1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
+                        1. Return _v_.
+                    1. [id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).
+                    1. [id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).
+                    1. Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).
+                    1. If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.
+                    1. Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).
+                    1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
+                    1. Return *+0*<sub>𝔽</sub>.
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-sortindexedproperties" type="abstract operation">
+                <h1>
+                  SortIndexedProperties (
+                    _obj_: an Object,
+                    _len_: a non-negative integer,
+                    _SortCompare_: an Abstract Closure with two parameters,
+                    <ins>_checkHasProperty_: a Boolean</ins>
+                  ): either a normal completion containing <del>an Object</del><ins>a List</ins> or an abrupt completion
+                </h1>
+                <dl class="header">
+                </dl>
+                <emu-alg>
+                  1. Let _items_ be a new empty List.
+                  1. Let _k_ be 0.
+                  1. Repeat, while _k_ &lt; _len_,
+                    1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                    1. <del>Let _kPresent_ be ? HasProperty(_obj_, _Pk_).</del>
+                    1. <ins>If _checkHasProperty_ is *true*, then</ins>
```

Comment:
this seems a bit awkward - it looks like `checkHasProperty` is set by the caller when the semantic they want is "no holes in the output"? maybe instead of naming the argument by how it works, it could be named by what it means, like "ignoreHoles" or something?

Reply:
agreed. Renamed to `ignoreHoles`